### PR TITLE
feat: http client applyCommit respects cache

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

**Review by oed — APPROVED** [review]
lgtm, thanks for adding tests to the http-client!

**Review by stbrody — APPROVED** [review]
LGTM

I'm curious how you found this?  Was this causing an actual issue somewhere?

**Comment by oed** [comment]
@stbrody  see https://github.com/ceramicstudio/js-3id-did-provider/pull/126#discussion_r626036420 but some of the conversation happened offline.

File: packages/http-client/src/__tests__/ceramic-http-client.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      expect(streamCache.size).toEqual(0);
+      const commit = {} as CeramicCommit;
+      await client.applyCommit(streamId, commit);
+      expect(streamCache.size).toEqual(1);
+      expect(streamCache.get(document.id.toString())).toBe(document);
+    });
+  });
+
+  describe('document not in cache', () => {
+    test('add to cache', async () => {
```
[the remark this answers is not in the document]
Comment (ukstv) [reply packages/http-client/src/__tests__/ceramic-http-client.test.ts]: Typo. Thanks.

**Pull request merged**